types/src/block/block_header/block_header_v2.rs

```diff
@@ -161,6 +165,11 @@ impl BlockHeaderV2 {
         self.era_end.is_some()
     }
 
+    /// Returns the public key of the validator which proposed the block.
+    pub fn proposer(&self) -> &PublicKey {
+        &self.proposer
+    }
+
```

Comment:
I thought we were also going to include the relevant last switch block hash in the header?

Reply:
We were I will add that now